eatngo-core/src/main/kotlin/com/eatngo/product/event/ProductEventListener.kt

```
@@ -38,12 +41,44 @@ class ProductEventListener(
+                localDate = today
+            )
+            inventoryCachePersistence.saveHash(orderItem.productId, InventoryDto.from(changedInventory))
+        }
         publishInventoryChangeIfNeeded(orderItem)
     }
 
     private fun processStockRollback(orderItem: OrderItem) {
-        inventoryCachePersistence.rollbackStock(orderItem.productId, orderItem.quantity)
+        try {
+            inventoryCachePersistence.rollbackStock(orderItem.productId, orderItem.quantity)
+        } catch (e: StockNotFound) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)